compiler/rustc_mir_build/src/build/coverageinfo.rs

```diff
@@ -378,4 +394,27 @@ impl Builder<'_, '_> {
             branch_info.record_conditions_operation(logical_op, span);
         }
     }
+
+    pub(crate) fn mcdc_increment_depth_if_enabled(&mut self) {
+        if let Some(branch_info) = self.coverage_branch_info.as_mut()
+            && let Some(mcdc_state) = branch_info.mcdc_state.as_mut()
+        {
+            mcdc_state.decision_ctx_stack.push(MCDCDecisionCtx::default());
```

Comment:
If the `decision_depth` varies in sync with `decision_ctx_stack`, could it be replaced with `decision_ctx_stack.len() - 1`?

Reply:
It could indeed. I'll just add a simple method around it, so it is not confusing to the reader.